src/components/productDetails/ProductDetails.tsx

```diff
@@ -11,12 +11,23 @@ import { Loader } from '../loader/Loader';
 import { ProductPrice } from '../products/price/ProductPrice';
 import { styles } from './Styles';
 
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
+import { MainStackParamList } from '../../navigation/stack-navigators/MainStackNavigator';
+import { StackNavigationProp } from '@react-navigation/stack';
+
 export const ProductDetails: React.FC = () => {
-    const { dataIsLoading, refresh, product } = useGetProduct(1);
+    const route = useRoute<RouteProp<MainStackParamList, 'ProductDetails'>>();
+    const navigation = useNavigation<StackNavigationProp<MainStackParamList>>();
+
+    const { dataIsLoading, refresh, product } = useGetProduct(route.params.productId);
+
+    useEffect(() => {
+        navigation.setOptions({
+            title: product && product.name ? product.name : 'Product Details',
```

Comment:
You may use ? operator from typescript to avoid additional checks  
```suggestion
            title: product?.name ? product.name : 'Product Details',
```

Reply:
Also I would propose to call setOptions only when data is already received. Right now it is called even on initial render. For initial route name you may use initialRouteName

```
if (product?.name) {
  navigation.setOptions({...
}
```